mantidproject/mantid
# Add input instrument approximation to MonteCarloAbsorption

This PR extends [`MonteCarloAbsorption`](http://docs.mantidproject.org/nightly/algorithms/MonteCarloAbsorption-v1.html) by adding an option to approximate the instrument of the input workspace by a 'sparse' instrument. The sparse instrument consists of a grid of detectors spawning the opening angles of the full instrument thus enabling simulation with smaller number of detectors than in the full input instrument. The simulated absorption corrections are then interpolated to the full instrument in a two step process.

This PR shouldn't introduce changes to existing functionality.

**To test:**

Run the algorithm by enabling and disabling *SparseInstrument* to see if the results more-or-less match. If possible, check that when *SparseInstrument* is disabled, the results are unchanged compared to master.

Fixes #19757.


*Release notes were updated accordingly.*


---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
